artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPLargeMessagePersister.java

```diff
@@ -96,10 +91,11 @@ public void encode(ActiveMQBuffer buffer, Message record) {
          properties.encode(buffer.byteBuf());
       }
 
-      ByteBuf savedEncodeBuffer = getSavedEncodeBuffer(record);
+      ByteBuf savedEncodeBuffer = msgEncode.getSavedEncodeBuffer();
       buffer.writeBytes(savedEncodeBuffer, 0, savedEncodeBuffer.writerIndex());
-      savedEncodeBuffer.release();
-      msgEncode.temporaryBuffer = null;
+      msgEncode.releaseEncodedBuffer();
+      msgEncode.releaseEncodedBuffer(); // We need two releases here, one for the retain when the savedEncodeBuffer was called
```

Comment:
Having 2 separate releases is prone to future errors: I believe this can be simplified to make esplicit in the method names what's happening here

Reply:
I will make a new method to do a release(2)

Reply:
Just to write here an example: https://netty.io/4.1/api/io/netty/buffer/ByteBuf.html#readRetainedSlice-int-

Reply:
I'm still fighting to find some water :( 
I would like you to be happy about the naming anyway, impl-wire you can use https://netty.io/4.1/api/io/netty/util/ReferenceCounted.html#release-int- that would release the buffer the number of times you need (and can return a boolean if you need to do something right after too)